.github/workflows/test-unit.yml

```diff
@@ -112,12 +110,11 @@ jobs:
 
       - name: Install PHP dependencies
         run: |
-          if [ "${{ matrix.type }}" != "Phpunit" ] && [ "${{ matrix.type }}" != "Phpunit Lowest" ] && [ "${{ matrix.type }}" != "Phpunit Burn" ]; then composer remove --no-interaction --no-update phpunit/phpunit --no-update phpunit/phpunit johnkary/phpunit-speedtrap --dev ; fi
+          if [ "${{ matrix.type }}" != "Phpunit" ] && [ "${{ matrix.type }}" != "Phpunit Lowest" ] ; then composer remove --no-interaction --no-update phpunit/phpunit --no-update phpunit/phpunit johnkary/phpunit-speedtrap --dev ; fi
           if [ "${{ matrix.type }}" != "CodingStyle" ]; then composer remove --no-interaction --no-update friendsofphp/php-cs-fixer --dev ; fi
           if [ "${{ matrix.type }}" != "StaticAnalysis" ]; then composer remove --no-interaction --no-update phpstan/phpstan --dev ; fi
           composer update --ansi --prefer-dist --no-interaction --no-progress --optimize-autoloader
           if [ "${{ matrix.type }}" == "Phpunit Lowest" ]; then composer update  --ansi --prefer-dist --prefer-lowest --prefer-stable --no-interaction --no-progress --optimize-autoloader ; fi
-          if [ "${{ matrix.type }}" == "Phpunit Burn" ]; then sed -i 's/ *public function runBare(): void/public function runBare(): void { gc_collect_cycles(); $mem0 = memory_get_usage(); for ($i = 0; $i < '"$(if [ \"$GITHUB_EVENT_NAME\" == \"schedule\" ]; then echo 5000; else echo 500; fi)"'; ++$i) { $this->_runBare(); if ($i === 0) { gc_collect_cycles(); $mem1 = memory_get_usage(); } } gc_collect_cycles(); $mem2 = memory_get_usage(); if ($mem2 - 2000 * 1024 > $mem0 || $mem2 - 256 * 1024 > $mem1) { $this->onNotSuccessfulTest(new AssertionFailedError("Memory leak detected! (" . round($mem0 \/ (1024 * 1024), 3) . " + " . round(($mem1 - $mem0) \/ (1024 * 1024), 3) . " + " . round(($mem2 - $mem1) \/ (1024 * 1024), 3) . " MB, " . $i . " iterations)")); } } private function _runBare(): void/' vendor/phpunit/phpunit/src/Framework/TestCase.php && cat vendor/phpunit/phpunit/src/Framework/TestCase.php | grep '_runBare(' ; fi
```

Comment:
what result did you get with x100 iterations?

if the leak is growing, just increase the hardcoded detection memory limits in the line above

Reply:
why not using this: https://github.com/Roave/no-leaks

i tried locally and the output is this :
 * Atk4\Login\Tests\AuthTest::testDb
 * Atk4\Login\Tests\AuthTest::testAuth
 * Atk4\Login\Tests\AuthTest::testAuthNoCache
 * Atk4\Login\Tests\Feature\PasswordManagementTest::testBasic
 * Atk4\Login\Tests\Feature\SendEmailActionTest::testBasic
 * Atk4\Login\Tests\Feature\SignupTest::testBasic
 * Atk4\Login\Tests\Feature\UniqueFieldValueTest::testBasic

I think is more specific, don't you think?

Reply:
I am for it if it will work on atk4/data and atk4/ui and detect all issues like variables assigned to static variables or DB connections left open.

My approach also tries to detect leaks without GC between consequent runs. Ideally, support should be added into https://github.com/Roave/no-leaks and make that like a regular phpunit plugin (to allow standard phpunit entry point).